Hymns/Common/SoundCloudWebView.swift

```diff
@@ -0,0 +1,66 @@
+import SwiftUI
+import WebKit
+
+struct SoundCloudWebView: UIViewRepresentable {
+    @Binding var isLoading: Bool  //Used for activity indicator
+    var url: URL
+    var loadStatusChanged: ((Bool, Error?) -> Void)?
+
+    func makeCoordinator() -> SoundCloudWebView.Coordinator {
+        Coordinator(self)
+    }
+
+    func makeUIView(context: Context) -> WKWebView {
+        let view = WKWebView()
+        view.navigationDelegate = context.coordinator
+        view.load(URLRequest(url: url))
+        return view
+    }
+
+    func updateUIView(_ uiView: WKWebView, context: Context) {
+    }
+
+    func onLoadStatusChanged(perform: ((Bool, Error?) -> Void)?) -> some View {
+        var copy = self
+        copy.loadStatusChanged = perform
+        return copy
+    }
+
+    class Coordinator: NSObject, WKNavigationDelegate {
+        let parent: SoundCloudWebView
+
+        init(_ parent: SoundCloudWebView) {
+            self.parent = parent
+        }
+
+        func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
+            parent.loadStatusChanged?(true, nil)
+        }
+
+        func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+
+            parent.isLoading = false  //Used for activity indicator
+            webView.evaluateJavaScript("document.querySelector('.app__upsell').remove();", completionHandler: { (_, _) -> Void in
+            })
+//            webView.evaluateJavaScript("document.querySelector('.app__header').remove();", completionHandler: { (_, _) -> Void in
+//            })
+            webView.evaluateJavaScript("document.querySelector('.app__footerPanel').remove();", completionHandler: { (_, _) -> Void in
+            })
+
+            webView.evaluateJavaScript("document.querySelector('.footer.show').remove();", completionHandler: { (_, _) -> Void in
+            })
+
+            webView.evaluateJavaScript("document.querySelector('.footer').remove();", completionHandler: { (_, _) -> Void in
+            })
+
+            webView.evaluateJavaScript("document.querySelector('.categories').remove();", completionHandler: { (_, _) -> Void in
+            })
```

Comment:
What do all these others things do? It looks like the .app__upsell is the only thing that we want to remove right? Or is there other stuff too?

Reply:
The other stuff is nice to remove. Without removing it we only get to see like the top 5 songs searched and then we get a big footer banner with another upsell thing saying to see more songs you need to use the app